File: 3719-longest-balanced-subarray-i/3719-longest-balanced-subarray-i.py
class Solution:
    def longestBalanced(self, nums: List[int]) -> int:
        ans = 0
        for i in range(len(nums)):
            odd_set = set(); even_set = set()
            for j in range(i, len(nums)):
                if nums[j] % 2: odd_set.add(nums[j])
                else: even_set.add(nums[j])
                if len(odd_set) == len(even_set):
                    ans = max(ans, j - i + 1)

        return ans
         
            
        
        

File: 3719-longest-balanced-subarray-i/3719-longest-balanced-subarray-i.c
int longestBalanced(int* nums, int numsSize) {
    int check[100001] = {0, };
    int even_len, odd_len; 
    int ans = 0;
    for (int i = 0; i < numsSize; i++) {
        memset(check, 0, 100001 * sizeof(int));
        even_len = 0;
        odd_len = 0; 
        for (int j = i; j < numsSize; j++) {
            int n = nums[j];
            if (check[n] == 0) {
                check[n] = 1;
                if (n % 2 == 1) {
                    odd_len += 1;
                } else {
                    even_len += 1;
                }
            }
            if (odd_len == even_len) {
                ans = (ans < j - i + 1) ? j - i + 1 : ans;
            }
        }
    }
    return ans;
}